deployments/hadoop-yarn/ansible/38-install-user-db.yml

```diff
@@ -0,0 +1,150 @@
+  hosts: zeppelin
+  become: yes
+  become_user: root
+  gather_facts: false
+  vars_files:
+    - config/ansible.yml
+    - config/hadoop.yml
+    - /tmp/ansible-vars.yml
+    - config/zeppelin.yml
+  vars:
+    zeppelinshiro: |
+            [main]
+            ds = com.mysql.cj.jdbc.MysqlDataSource
+            ds.serverName = localhost
+            ds.databaseName = zeppelin
+            ds.user = zeppelin
+            ds.password = {{ mysql_zeppelin_password }}
+
+            jdbcRealm = org.apache.shiro.realm.jdbc.JdbcRealm
+            ps = org.apache.shiro.authc.credential.DefaultPasswordService
+            pm = org.apache.shiro.authc.credential.PasswordMatcher
+            pm.passwordService = $ps
+
+            jdbcRealm.dataSource = $ds
+            jdbcRealm.credentialsMatcher = $pm
+            shiro.loginUrl = /api/login
+
+            sessionManager = org.apache.shiro.web.session.mgt.DefaultWebSessionManager
+            cookie = org.apache.shiro.web.servlet.SimpleCookie
+            cookie.name = JSESSIONID
+            cookie.httpOnly = true
+            sessionManager.sessionIdCookie = $cookie
+            securityManager.sessionManager = $sessionManager
+            securityManager.sessionManager.globalSessionTimeout = 86400000
+
+
+            [urls]
+            /** = authc
+            /api/version = anon
+            /api/interpreter/setting/restart/** = authc
+            /api/interpreter/** = authc, roles[admin]
+            /api/configurations/** = authc, roles[admin]
+            /api/credential/** = authc, roles[admin]
+
+            [roles]
+            user = *
+            admin = *
+
+  tasks:
+
+   - set_fact:
+       mysql_root_password: "{{ lookup('password','/dev/null chars=ascii_letters,digits,hexdigits length=20') }}!"
+       mysql_zeppelin_password: "{{ lookup('password','/dev/null chars=ascii_letters,digits,hexdigits length=20') }}!"
+
+   - name: "Enable MySQL Community release repo"
+     yum:
+       disable_gpg_check: True
+       name: https://repo.mysql.com//mysql80-community-release-fc31-1.noarch.rpm
+       state: present
+
+   - name: "Install MySQL Server"
+     yum: name=mysql-server state=installed
+
+   - name: "Install MySQL-devel"
+     yum: name=mysql-devel state=installed
+
+   - name: "Make sure pymysql is present"
+     pip:
+       name: pymysql
+       state: present
+
+   - name: "Start the MySQL service"
+     service: 
+       name: mysqld 
+       state: started
+       enabled: yes
+
+   - name: "Find temporary password"
+     shell: "echo `grep 'temporary.*root@localhost' /var/log/mysqld.log | sed
+'s/.*root@localhost: //'`"
+     register: mysql_root_password_temp
+     tags: register
+
+   - name: "Update expired root user password"
+     shell: 'mysql -e "ALTER USER ''root''@''localhost'' IDENTIFIED WITH mysql_native_password BY ''{{ mysql_root_password }}''" --connect-expired-password -uroot -p"{{ mysql_root_password_temp.stdout }}"'
```

Comment:
I think this is simpler for the version of MariaDB that is installed by the OS packages. From what I've read MariaDB allows root access to the database via a local Unix socket from by the `root` user account. So it should be possible to login to the database as root with no password and then set the password for subsequent access.